AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/IgnoreWhenHandler.java

```diff
@@ -53,6 +59,21 @@ public void process(Element element, EFragmentHolder holder) throws Exception {
 		JMethod delegatingMethod = codeModelHelper.overrideAnnotatedMethod(executableElement, holder);
 		JBlock previousMethodBody = codeModelHelper.removeBody(delegatingMethod);
 
-		delegatingMethod.body()._if(invoke(holder.getGeneratedClass().staticRef("this"), "getActivity").ne(_null()))._then().add(previousMethodBody);
+		IgnoreWhen ignoreWhen = element.getAnnotation(IgnoreWhen.class);
+		switch (ignoreWhen.value()) {
+		case VIEW_DESTROYED:
+			String viewDestroyedFieldName = "viewDestroyed" + generationSuffix();
+			JFieldVar viewDestroyed = holder.getGeneratedClass().fields().get(viewDestroyedFieldName);
+			if (viewDestroyed == null) {
+				viewDestroyed = holder.getGeneratedClass().field(PRIVATE | VOLATILE, getCodeModel().BOOLEAN, viewDestroyedFieldName);
+				holder.getSetContentViewBlock().assign(viewDestroyed, FALSE);
+				holder.getOnDestroyViewAfterSuperBlock().assign(viewDestroyed, TRUE);
+			}
+			delegatingMethod.body()._if(viewDestroyed.not())._then().add(previousMethodBody);
```

Comment:
Let's remove code duplication here: only the conditional expression depends on the lifecycle state, the other code is the same in both cases.